assets/scripts/admin/surveysettings.js

```diff
@@ -152,6 +152,40 @@ function guidGenerator() {
     return (S4() + S4() + '-' + S4() + '-' + S4() + '-' + S4() + '-' + S4() + S4() + S4());
 }
 
+/**
+ * Validate settings form depending on form name
+ */
+function validateSettingsForm($form) {
+    switch ($form.attr('id')) {
+        case 'publication':
+            return validateEndDateHigherThanStart(
+                $('#startdate_datetimepicker').data('DateTimePicker'),
+                $('#expires_datetimepicker').data('DateTimePicker'),
+                expirationLowerThanStartError
+            );
+        default:
+            return true;
+    }
+}
+
+/**
+ * Validates that an end date is not lower than a start date
+ */
+function validateEndDateHigherThanStart(startDatePicker, endDatePicker, errorMessage) {
+    if (!startDatePicker || !startDatePicker.date()) {
+        return true;
+    }
+    if (!endDatePicker || !endDatePicker.date()) {
+        return true;
+    }
+    const difference = endDatePicker.date().diff(startDatePicker.date());
+    if (difference >= 0) {
+        return true;
+    }
+    LS.LsGlobalNotifier.createFlash(errorMessage, 'alert-danger fade in');
```

Comment:
Preferably, you shouldn't mix pure logic code (validate date) with effectful code (show message), but I'll allow it for now. :) You want to unit test the logic code, but integrity test the effectful code, this is the main reason to keep it separated. Also, it creates higher composability of the code base parts.